# Vaihe 1

### Kiinnostavat tutkimuskysymykset
1. Mitkä genret ovat olleet suosittuja milloin?
2. Mikä näyttelijöiden yhdistelmä tuottaa suosittuja elokuvia?
3. Korrelaatio arvioiden ja tuoton välillä. Voiko elokuva olla huonosti arvioitu mutta silti tuottoisa tai toisin päin?
4. Jos tekisimme elokuvan, millainen sen kannattaa olla, että se on tuottoisa ja suosittu?

# Vaihe 2

### Aineiston kuvaus
Aineisto löydettiin Kagglesta ja se sisältää tietoa tuhannesta parhaiten arvioidusta elokuvasta ja TV -sarjasta IMDB palvelussa.

Linkki aineistoon: [LINKKI](https://www.kaggle.com/datasets/ashpalsingh1525/imdb-movies-dataset)


In [60]:
import numpy as np
import pandas as pd
# set display options
pd.set_option('display.max_colwidth', None)

dataset = pd.read_csv('Datasets/imdb_movies.csv')

# TODO: Fixaa tää silleen et selitetään otsikoiden tarkoitus
# Print dataframe columns
print("Aineiston otsikot:")
print(dataset.columns)
print(dataset.shape)


Aineiston otsikot:
Index(['names', 'date_x', 'score', 'genre', 'overview', 'crew', 'orig_title',
       'status', 'orig_lang', 'budget_x', 'revenue', 'country'],
      dtype='object')
(10178, 12)


Aineiston otsikot:

TODO: Lisää selitys otiskoista Crew Members otsikko on tosi outo, se pitää selittää jotenkin


Aineistossa on siis 12 otsikkoa ja 10178 havaintoa.


Osa aineiston datasta on epäilyttävää. Esimerkiksi 144 elokuvista on saanut 0.0 arvostelun, ja niiden kaikkien tuotto on tasan $175,269,998.80

Aineiston julkaisija on ilmeisesti käyttänyt jotain knn keinoa aineiston täyttämiseen. Tämä täytyy ottaa huomioon myöhemmin.



# Vaihe 3
## Aineiston esikäsittely

#### Tyhjien havaintojen poisto

In [ ]:
# Tässä poistetaan ne havainnot joissa on tyhjiä arvoja tai arvostelu on 0.0
dataset = dataset.dropna()
dataset = dataset[dataset['score'] != 0.0]
print(dataset.shape)

# Poistetaan elokuvat, jotka ei ole julkaistu
dataset = dataset['status'] = "Released"

(9889, 12)


Aineistosta poistettiin 289 havaintoa

#### Avataan näyttelijät

In [62]:
print("Aineiston näyttelijät:")
print(dataset['crew'].head(1))

Aineiston näyttelijät:
0    Michael B. Jordan, Adonis Creed, Tessa Thompson, Bianca Taylor, Jonathan Majors, Damien Anderson, Wood Harris, Tony 'Little Duke' Evers, Phylicia Rashād, Mary Anne Creed, Mila Davis-Kent, Amara Creed, Florian Munteanu, Viktor Drago, José Benavidez Jr., Felix Chavez, Selenis Leyva, Laura Chavez
Name: crew, dtype: object


In [63]:
# Tässä avataan näyttelijät. Näyttelijän hahmo on listattu näyttelijän jälkeen. (esim. Tom Cruise, Ethan Hunt, ...)
dataset['crew'] = dataset['crew'].str.split(', ')
# Poista joka toinen alkio
dataset['crew'] = dataset['crew'].apply(lambda x: x[::2])
print("1. elokuvan näyttelijät:")
print(dataset.head(1))

1. elokuvan näyttelijät:
       names       date_x  score          genre  \
0  Creed III  03/02/2023    73.0  Drama, Action   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                     overview  \
0  After dominating the boxing world, Adonis Creed has been thriving in both his career and family life. When a childhood friend and former boxing prodigy, Damien Anderson, resurfaces after serving a long sentence in prison, he is eager to prove that he deserves his shot in the ring. The face-off between former friends is more than just a fight. To settle the score, Adonis must put his future on the line

# Vaihe 4
### Aineiston analysointi

In [64]:
#  JOKAISESSA ANALYSOINTIKOHDASSA TEHDÄÄN AINEISTOSTA KOPIO, ETTEI MYÖHEMMÄT KOHDAT MUUTU

#### Mitkä genret ovat suosittuja milloin

#### Mikä näyttelijöiden yhdistelmä tuottaa suosittuja elokuvia?

#### Korrelaatio arvioiden ja tuoton välillä. Voiko elokuva olla huonosti arvioitu mutta silti tuottoisa tai toisin päin?

#### Jos tekisimme elokuvan, millainen sen kannattaa olla, että se on tuottoisa ja suosittu?

# Vaihe 5
### Analyysin tulkinta

# Vaihe 6
### Tulosten hyödyntäminen